src/reactapp/src/components/CheckoutForm.jsx

```diff
@@ -15,20 +15,37 @@ import StickyRightSidebar from './StickyRightSidebar';
 import CheckoutAgreements from './checkoutAgreements';
 import CheckoutFormWrapper from './CheckoutFormWrapper';
 import { config } from '../config';
+import { aggregatedQueryRequest } from '../api';
 import useAppContext from '../hook/useAppContext';
 import useCartContext from '../hook/useCartContext';
 
 function CheckoutForm() {
-  const { orderId, getGuestCartInfo } = useCartContext();
-  const [{ pageLoader }, { setPageLoader }] = useAppContext();
+  const { orderId, storeAggregatedCartStates } = useCartContext();
+  const [{ pageLoader }, appActions] = useAppContext();
+  const {
+    setPageLoader,
+    dispatch: appDispatch,
+    storeAggregatedAppStates,
+  } = appActions;
 
   useEffect(() => {
     (async () => {
-      setPageLoader(true);
-      await getGuestCartInfo();
-      setPageLoader(false);
+      try {
+        setPageLoader(true);
+        const data = await aggregatedQueryRequest(appDispatch);
```

Comment:
Your thought is in the right way. But we have an ambiguous situation here as we need to decide in which context we will put this action. It collects both App and Cart information. Depending on both App and Cart information, we are mutating the Formik context too. So in short, the `aggregatedQueryRequest` belongs to App, Cart and Formik context.

So using it directly makes much more sense here eventhough we deroute from the traditional approach!

Reply:
This seems like a good time to introduce an AggregateContext, the unique dispatch can then be used to call `App/aggregated/actions.js` and `Cart/aggregated/actions.js` functions too. You wouldnt have the `data` variable in the react component and could remove the store functions too. Additionally you can guarantee this action method is called only a single time by providing an empty array to the `useEffect` hook. An empty array means the `useEffect` function is only ran after the first render.

Reply:
Feel like an over-kill. Because `AggregateContext` will only be used in this occasion only. We cannot move `App/aggregated/actions.js` and `Cart/aggregated/actions.js` into `AggregateContext`. Because the dispatches of those actions are different. We cannot use AggregateDispatch to update Cart or App states.

Reply:
Ok, I think I get what you mean the dispatch wouldnt update the intended reducers.. I guess the current approach is what will have to do